ksql-streams/src/main/java/io/confluent/ksql/execution/streams/StreamSourceBuilder.java

```diff
@@ -47,55 +49,73 @@ private StreamSourceBuilder() {
       final KsqlQueryBuilder queryBuilder,
       final StreamSource streamSource
   ) {
+    final KeyFormat keyFormat = streamSource.getFormats().getKeyFormat();
```

Comment:
Slight refactoring of this method as I couldn't grok it. (Apologies!)

Added a sanity test that it is being called correctly, i.e. with a non-windowed keyed stream.